lab-khalid/route/auth-router.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+const
+  Router = require('express').Router,
+  jsonParser = require('body-parser').json(),
+  debug = require('debug')('CFgram:auth-router'),
+  authmiddleware = require('../lib/auth-middleware.js'),
+
+  User = require('../model/user.js'),
+
+  authRouter = module.exports = new Router();
+
+authRouter.post('/api/signup',jsonParser, function(req, res, next){
+  debug('POST /api/signup');
+  let password = req.body.password;
+  delete req.body.password;
+
+  let user = new User(req.body);
+  user.genpasswordHash(password)
```

Comment:
this should be `.genPasswordHash` because password is capitalized in line 21 of user.js